ml/create_training_dataset.sh

```diff
@@ -7,46 +7,67 @@ source utils/setup_cvmfs_sft.sh
-then
-    ARTUS_FRIENDS=${ARTUS_FRIENDS_MT}
-fi
-if [ ${CHANNEL} == 'et' ]
-then
-    ARTUS_FRIENDS=${ARTUS_FRIENDS_ET}
-fi
-if [ ${CHANNEL} == 'tt' ]
-then
-    ARTUS_FRIENDS=${ARTUS_FRIENDS_TT}
-fi
-if [ ${CHANNEL} == 'em' ]
-then
-    ARTUS_FRIENDS=${ARTUS_FRIENDS_EM}
-fi
-
-# Write dataset config
-python ml/write_dataset_config.py \
-    --era ${ERA} \
-    --channel ${CHANNEL} \
-    --base-path $ARTUS_OUTPUTS \
-    --friend-paths $ARTUS_FRIENDS \
-    --database $KAPPA_DATABASE \
-    --output-path $PWD/ml/${ERA}_${CHANNEL} \
-    --output-filename training_dataset.root \
-    --tree-path ${CHANNEL}_nominal/ntuple \
-    --event-branch event \
-    --training-weight-branch training_weight \
-    --output-config ml/${ERA}_${CHANNEL}/dataset_config.yaml
-
-# Create dataset files from config
-./htt-ml/dataset/create_training_dataset.py ml/${ERA}_${CHANNEL}/dataset_config.yaml
-
-# Reweight STXS stage 1 signals so that each stage 1 signal is weighted equally but
-# conserve the overall weight of the stage 0 signal
-#python ml/reweight_stxs_stage1.py \
-#    ml/${ERA}_${CHANNEL} \
-#    ml/${ERA}_${CHANNEL}/fold0_training_dataset.root \
-#    ml/${ERA}_${CHANNEL}/fold1_training_dataset.root
+function run_procedure() {
+    SELERA=$1
+    SELCHANNEL=$2
+    mkdir -p ml/${SELERA}_${SELCHANNEL}
+
+    ARTUS_FRIENDS=""
+    if [ ${SELCHANNEL} == 'mt' ]
+    then
+        ARTUS_FRIENDS=${ARTUS_FRIENDS_MT}
+    fi
+    if [ ${SELCHANNEL} == 'et' ]
+    then
+        ARTUS_FRIENDS=${ARTUS_FRIENDS_ET}
+    fi
+    if [ ${SELCHANNEL} == 'tt' ]
+    then
+        ARTUS_FRIENDS=${ARTUS_FRIENDS_TT}
+    fi
+    if [ ${SELCHANNEL} == 'em' ]
+    then
+        ARTUS_FRIENDS=${ARTUS_FRIENDS_EM}
+    fi
+
+    # Write dataset config
+    python ml/write_dataset_config.py \
+        --era ${SELERA} \
+        --channel ${SELCHANNEL} \
+        --base-path $ARTUS_OUTPUTS \
+        --friend-paths $ARTUS_FRIENDS \
+        --database $KAPPA_DATABASE \
+        --output-path $PWD/ml/${SELERA}_${SELCHANNEL} \
+        --output-filename training_dataset.root \
+        --tree-path ${SELCHANNEL}_nominal/ntuple \
+        --event-branch event \
+        --training-weight-branch training_weight \
+        --training-z-estimation-method emb \
+        --output-config ml/${SELERA}_${SELCHANNEL}/dataset_config.yaml
+
+    # Create dataset files from config
+    ./htt-ml/dataset/create_training_dataset.py ml/${SELERA}_${SELCHANNEL}/dataset_config.yaml
+
+    # Reweight STXS stage 1 signals so that each stage 1 signal is weighted equally but
+    # conserve the overall weight of the stage 0 signal
+    #python ml/reweight_stxs_stage1.py \
+    #    ml/${SELERA}_${SELCHANNEL} \
+    #    ml/${SELERA}_${SELCHANNEL}/fold0_training_dataset.root \
+    #    ml/${SELERA}_${SELCHANNEL}/fold1_training_dataset.root
+}
+
+CHANNELS=$(if [[ -z $CHANNEL ]]; then echo "tt" "mt" "et"; else echo $CHANNEL; fi)
+ERAS=$(if [[ -z $ERA ]]; then echo "2016" "2017"; else echo $ERA; fi)
```

Comment:
To be most general, please also add `"2018"`

Reply:
done